edi_oca/models/edi_exchange_record.py

```diff
@@ -453,11 +461,21 @@ def _search(
         for model, targets in model_data.items():
             if not self.env[model].check_access_rights("read", False):
                 continue
-            target_ids = list(targets)
+            recs = self.env[model].browse(list(targets))
```

Comment:
Just a remark: as this runs on every search it might add overhead when working on large datasets.